src/main/java/mate/academy/internetshop/dao/GenericDao.java

```diff
@@ -0,0 +1,18 @@
+package mate.academy.internetshop.dao;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface GenericDao<T, N> {
+    T create(T entity);
+
+    Optional<T> get(N entityN);
```

Comment:
```suggestion
    Optional<T> get(ID id);
```
не принципово, але думаю, так зрозуміліше